features/cellular/easy_cellular/EasyCellularConnection.cpp

```diff
@@ -128,8 +132,15 @@ void EasyCellularConnection::set_credentials(const char *apn, const char *uname,
 
 void EasyCellularConnection::set_sim_pin(const char *sim_pin)
 {
-    if (sim_pin) {
-        _cellularConnectionFSM.set_sim_pin(sim_pin);
+    if (sim_pin && strlen(sim_pin) > 0) {
+        if (!_cellularConnectionFSM) {
+            _credentials_err = init();
+
+            if (_credentials_err) {
+                return;
```

Comment:
should we just assert here..?

Reply:
_credentials_err is used later when trying to connect to return proper error. Because of the inherited void function we must store error for later use.